patches/tModLoader/Terraria/Player_tML.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Terraria.ModLoader;
+
+namespace Terraria
+{
+	public partial class Player
+	{
+		private DamageClassData[] damageData = new DamageClassData[0];
+
+		internal void ResetDamageClassDictionaries() {
+			damageData = new DamageClassData[DamageClassLoader.DamageClassCount];
+			for (int i = 0; i < DamageClassLoader.DamageClassCount; i++) {
+				damageData[i] = new DamageClassData(4, 0, 1);
+			}
+		}
+
+		/// <summary>
+		/// Edits the crit for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount that should be added to the crit. Can be negative too.</param>
+		public void ModifyCrit<T>(int changeAmount) where T : ModDamageClass => ModifyCrit(ModContent.GetInstance<T>(), changeAmount);
+
+		/// <summary>
+		/// Edits the damage value for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount (as a percentage, 0.01f is 1%) that should be added to the damage. Can be negative too.</param>
+		public void ModifyDamage<T>(float changeAmount) where T : ModDamageClass => ModifyDamage(ModContent.GetInstance<T>(), changeAmount);
+
+		/// <summary>
+		/// Edits the damage multiplier's value for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount (as a percentage, 0.01f is 1%) that should be added to the damage multiplier. Can be negative too.</param>
+		public void ModifyDamageMult<T>(float changeAmount) where T : ModDamageClass => ModifyDamageMult(ModContent.GetInstance<T>(), changeAmount);
+
+		/// <summary>
+		/// Edits the crit for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount that should be added to the crit. Can be negative too.</param>
+		public void ModifyCrit(ModDamageClass damageClass, int changeAmount) {
+			if (damageClass is Summon) {
+				return;
+			}
+			ref var data = ref damageData[damageClass.index];
+			data.crit += changeAmount;
+		}
+
+		/// <summary>
+		/// Edits the damage value for the given damage type on this player.
+		/// </summary>
+		/// <param name="changeAmount">The amount (as a percentage, 0.01f is 1%) that should be added to the damage. Can be negative too.</param>
+		public void ModifyDamage(ModDamageClass damageClass, float changeAmount) {
```

Comment:
I think all these should have Add in place of Modify. Though this will be affected when you merge CB's PR.